main.go

```diff
@@ -209,6 +221,24 @@ func main() {
 	dib.Close()
 }
 
+func setupDiscordFormat(discordFormat map[string]string) map[string]string {
```

Comment:
I think it would be good to return an error if an unknown key is present in the map.

- If it's the first load, we can use the error to fatally exit: `log.WithError(err).Fatal("discord_format setting is invalid")`
- If it's a live reload, use the error to print an error message (`log.WithError(err).Error("discord_format setting is invalid, this setting has not been updated")`), and ignore the change (don't assign to `dib.Config.DiscordFormat`)

Reply:
You mean log that format setting is invalid and if there isn't an old format setting use default but if there is use the old format?

Reply:
Kind of, I meant changing the return type of `setupDiscordFormat` to `(map[string]string, error)`. And if there's no old format setting, i.e. at the very start, prevent the bot from starting (hence, "fatal").

Reply:
Hmmm, fatal first load exit. This wouldn't allow for the `show_joinquit` to be supplanted though and for a default format of `" "` to basically be "disable" and anything else to be enable that line. Let me know what you think you'd really prefer here, because if #72 and #79 get merged then `show_joinquit` doesn't exactly make sense as the configuration setting name any longer and being able to granularly enable/disable what events you want to write towards Discord might be a good thing.

Reply:
This would only be if an unknown key is present in the map, i.e. if you accidentally type "quiiit" instead of "quit".

Reply:
Ah, that would be fine I think, I'll see about detecting unknown keys and propagating an error.

Reply:
Have added code that should do basically what is mentioned here.